controllers/image/api/v1/image_types.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ImageKeyword todo add image keywords
+// +kubebuilder:validation:Enum={calculation, database, operator, unknown}
+// +kubebuilder:default=unknown
+type ImageKeyword string
+
+const (
+	ImageKeywordCalculation ImageKeyword = "calculation"
+	ImageKeywordDatabase    ImageKeyword = "database"
+	ImageKeywordOperator    ImageKeyword = "operator"
+	ImageKeywordUnknown     ImageKeyword = "unknown"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type ImageBaseInfo struct {
+	//+kubebuilder:validation:Required
+	//+kubebuilder:validation:MaxLength=1024
+	Name string `json:"name"`
+
+	//+kubebuilder:validation:Required
+	Tags []string `json:"tags"`
+}
+
+type ImageDetailInfo struct {
+	URL         string         `json:"url,omitempty"`
+	Keywords    []ImageKeyword `json:"keywords,omitempty"`
+	Description string         `json:"description,omitempty"`
+	Icon        string         `json:"icon,omitempty"`
+	Docs        string         `json:"docs,omitempty"`
```

Comment:
What if every version has diff docs and description?

Reply:
So Documents and descriptions need to be synchronized with the image name + image tag.
Refer to this [issue](https://github.com/labring/sealos/issues/1902).